### PR TITLE
feat: [PL-27391]: Adding Custom SM support in secret flow

### PR DESCRIPTION
**Pull request by meenakshi-harness**
### Summary
All changes related to Custom Secret Manager are reviewed in https://github.com/harness/harness-core-ui/pull/10467
For Secret Flow : https://github.com/harness/harness-core-ui/pull/10657/commits/df1fe7e8aec99140606276426a5dbbc0585466ea

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>
Screenshot 2022-08-19 at 9 01 24 PM

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 591 Issues 
* Code issue 591 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10657)

**Comment by meenakshi-harness** [comment]
retrigger coverage

**Comment by meenakshi-harness** [comment]
All changes related to Custom Secret Manager are reviewed in https://github.com/harness/harness-core-ui/pull/10467
For Secret Flow : https://github.com/harness/harness-core-ui/pull/10657/commits/ba589136163ba8e319b0909b51b7567fe907955f

**Comment by meenakshi-harness** [comment]
retrigger coverage

**Comment by meenakshi-harness** [comment]
retrigger typecheck

**Comment by meenakshi-harness** [comment]
retrigger titlecheck

**Review by upadhyay-an — APPROVED** [review]

**Review by pbarapatre10 — APPROVED** [review]
LGTM

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx [hunk 1]
```diff
@@ -300,6 +310,23 @@ const CreateUpdateSecret: React.FC<CreateUpdateSecretProps> = props => {
 
   const [selectedSecretManager, setSelectedSecretManager] = useState<ConnectorInfoDTO | undefined>()
   const [readOnlySecretManager, setReadOnlySecretManager] = useState<boolean>()
+  const [templateInputSets, setTemplateInputSets] = React.useState<JsonNode>()
+
+  const initializeTemplateInputs = (secretManager: ConnectorInfoDTO | undefined) => {
+    if (secretManager?.type === 'CustomSecretManager') {
+      const inputs: [] = secretManager.spec.template?.templateInputs?.environmentVariables
+      if (inputs) {
```
Comment (swarajcheguri-harness) [inline src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:318]: should there be a length check too here ?

**Review by swarajcheguri-harness — APPROVED** [review]

File: src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx (hunk 1 above)
Comment (meenakshi-harness) [reply src/modules/30-secrets/components/CreateUpdateSecret/CreateUpdateSecret.tsx:318]: No the child component will handle that case.

**Pull request merged**